notion_client/lib/api_objects.py

```diff
@@ -0,0 +1,430 @@
+    def _from_json(cls, d: Dict[str, Any]) -> "PropertyValue":
+        required_fields = cls.__dataclass_fields__  # type: ignore
+
+        # Try converting all json datatypes to their dataclass type (e.g Enum).
+        # Property subclasses with more complicated data types will
+        # have to override `from_json`
+        for k in required_fields.keys():
+            if d.get(k) is not None:
+                d[k] = required_fields[k].type(d.get(k))
+
+        result: Dict[str, Any] = dict([(k, d.get(k)) for k in required_fields.keys()])
+        return cls(**result)
+
+    @classmethod
+    def from_json(cls, d: Dict[str, Any]) -> "PropertyValue":
+        property_id = d.pop("id")
+        property_type = d.pop("type")
+
+        property_value = cls._from_json(d)
+        property_value.property_id = property_id
+        property_value.property_type = property_type
+        return property_value
+
+
+@dataclass
+class RollupPropertyElement(PropertyValue):
+    pass
+
+
+@dataclass
+class FormulaPropertyValue(PropertyValue):
+    type: str
+    string: str
+    number: float
+    boolean: bool
+    date: datetime
+    relation: List[PageReference]
+
+    @classmethod
+    def _from_json(cls, d: Dict[str, Any]) -> "FormulaPropertyValue":
+        d = {**d, **d[PropertyType.formula.value]}
+        date = None
+        relations = None
+        if d.get("date"):
+            date = d.pop("date")
+
+        if d.get("relation"):
+            relations = d.pop("relations")
+
+        formula: FormulaPropertyValue = super(cls, FormulaPropertyValue)._from_json(d)
+
+        if relations:
+            formula.relation = [PageReference(x["id"]) for x in relations]
+
+        if date:
+            formula.date = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        return formula
+
+
+@dataclass
+class RelationPropertyValue(PropertyValue):
+    database_id: str
+    synced_property_name: str
+    synced_property_id: str
+
+
+@dataclass
+class RollupPropertyValue(PropertyValue):
+    type: RollupValueTypes
+    number: float
+    date: datetime
+    array: List[RollupPropertyElement]
+
+    @classmethod
+    def _from_json(cls, d: Dict[str, Any]) -> "RollupPropertyValue":
+        d = {**d, **d[PropertyType.rollup.value]}
+        date = None
+        array = None
+        if d.get("date"):
+            date = d.pop("date")
+        if d.get("array"):
+            array = d.pop("array")
+
+        rollup: RollupPropertyValue = super(cls, RollupPropertyValue)._from_json(d)
+
+        if array:
+            arrays: List[RollupPropertyElement] = [
+                RollupPropertyElement.from_json(r) for r in array  # type: ignore
+            ]
+            rollup.array = arrays
+        if date:
+            rollup.date = datetime.strptime(date, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        return rollup
+
+
+@dataclass
+class NumberPropertyValue(PropertyValue):
+    number: float
+
+
+@dataclass
+class SelectPropertyValue(PropertyValue):
```

Comment:
See previous comment about PropertyValue above.

This class is almost identical to SelectProperty and its use of SelectOption.